extension/background.js

```diff
@@ -1,112 +1,49 @@
-var windowCount = 'windowCount',
-	incogHistory = 'incogHistory',
-	incogRecent = 'incogRecent';
+if (!chrome.extension.inIncognitoContext) window.close();
 
-chrome.runtime.onMessage.addListener(function (request, sender, sendResponse) {
+var tabs = JSON.parse(localStorage.tabs || '{}'), incHist = JSON.parse(localStorage.incHist || '[]'), incRecent = JSON.parse(localStorage.incRecent || '[]');

```

Comment:
Use `getItem` method for accessing a value from localstorage.

And we can reuse old local storage values - `"incogHistory"`, `"incogRecent"` and for tabs we can use `"incogTabs"`.

Reply:
We have to tread carefully with these variables. When the inco background page is swapped (very frequently), the data structure is destroyed. And when it is restored, internal links between variables are not restored.
If you prefer to use `getItem`, no problem. I think I took a shortcut at the time of writing.  ;)
You can rename the variables. I tend to use very short names, but it can be confusing. Sometimes I even mix english abbreviations and french ones (not in this case).